windows/network_device_ops/network_adapter_status_check.yml

```diff
@@ -1,90 +1,89 @@
-    win_ping_src_ip: "{{ new_nic_ip_address | default('') }}"
-- name: Set fact of the ping vlan gateway result
+- name: "Save the ping gateway result"
   ansible.builtin.set_fact:
     ping_result_after_hotadd: "{{ win_ping_target_result }}"
 
-# Ping vlan gateway after network adapter disabled
-- include_tasks: ../utils/win_set_netadapter_status.yml
+- name: "Disable new added network adapter in guest OS"
+  include_tasks: ../utils/win_set_netadapter_status.yml
   vars:
     win_set_netadapter_operation: "Disabled"
-    win_set_netadapter_name: "{{ win_get_ethernet_name_result }}"
-- include_tasks: ../utils/win_ping_target.yml
+    win_set_netadapter_name: "{{ win_get_ethernet_name }}"
+
+- name: "Ping gateway after new network adapter disabled"
+  include_tasks: ../utils/win_ping_target.yml
   vars:
-    win_ping_target_ip: "{{ vlan_gateway }}"
-    win_ping_src_ip: "{{ new_nic_ip_address | default('') }}"
-- name: Set fact of the ping vlan gateway result after disable
+    win_ping_target_ip: "{{ network_env_gateway }}"
+    win_ping_src_ip: "{{ new_nic_ip_address }}"
+
+- name: "Save the ping gateway result"
   ansible.builtin.set_fact:
     ping_result_after_disable: "{{ win_ping_target_result }}"
 
-# Ping vlan gateway after network adapter enabled
-- include_tasks: ../utils/win_set_netadapter_status.yml
+- name: "Enable new added network adapter in guest OS"
+  include_tasks: ../utils/win_set_netadapter_status.yml
   vars:
     win_set_netadapter_operation: "Up"
-    win_set_netadapter_name: "{{ win_get_ethernet_name_result }}"
-- include_tasks: ../utils/win_ping_target.yml
+    win_set_netadapter_name: "{{ win_get_ethernet_name }}"
+
+- name: "Ping gateway after new network adapter enabled"
+  include_tasks: ../utils/win_ping_target.yml
   vars:
-    win_ping_target_ip: "{{ vlan_gateway }}"
-    win_ping_src_ip: "{{ new_nic_ip_address | default('') }}"
-- name: Set fact of the ping vlan gateway result after re-enable
+    win_ping_target_ip: "{{ network_env_gateway }}"
+    win_ping_src_ip: "{{ new_nic_ip_address }}"
+
+- name: "Save the ping gateway result"
   ansible.builtin.set_fact:
     ping_result_after_enable: "{{ win_ping_target_result }}"
 
-# Ping vlan gateway after network adapter disconnected
-- include_tasks: ../../common/vm_configure_network_adapter.yml
+- name: "Disconnect new network adapter on VM"
+  include_tasks: ../../common/vm_configure_network_adapter.yml
   vars:
     netadapter_mac_addr: "{{ new_nic_mac_address }}"
     netadapter_connect: false
-- name: Check disconnect network adapter to VM succeed
+
+- name: "Check disconnect new network adapter operation"
   ansible.builtin.assert:
     that:
+      - reconfig_result.changed is defined
       - reconfig_result.changed
-    fail_msg: "Disconnect VM network adapter task result is not changed."
-- include_tasks: ../utils/win_ping_target.yml
+    fail_msg: "The result of VM network adapter disconnection task is not changed."
```

Comment:
If there is an error in above `vm_configure_network_adapter.yml` task, will the error be printed in `reconfig_result.msg`? If yes, I think here can append the error message like `{{ reconfig_result.msg | default('') }}"

Reply:
If there is error then above task will fail normally, here check 'changed' to make sure the above task changes the VM config actually.